includes/class-amp-theme-support.php

```diff
@@ -2376,6 +2318,69 @@ public static function prepare_response( $response, $args = [] ) {
 		return $response;
 	}
 
+	/**
+	 * Optimizer instance to use.
+	 *
+	 * @param array $args Associative array of arguments to pass into the transformation engine.
+	 * @return Optimizer\TransformationEngine Optimizer transformation engine to use.
+	 */
+	private static function get_optimizer( $args ) {
+		$configuration = self::get_optimizer_configuration( $args );
+
+		// @todo Replace CurlRemoteRequest with a Requests version?
```

Comment:
We do currently require cURL, but only for the sake of FasterImage:

https://github.com/ampproject/amp-wp/blob/88f47ce88887156de30c1d63d6ca43150badfd63/amp.php#L44-L53

We should make sure that `curl_getinfo()` is added to the list as it is used by `CurlRemoteRequest`.

Otherwise, if not for FasterImage, perhaps this should be using the `WP_HTTP` interface? It supports both `curl` and `streams` as transports.

Reply:
I can add a `WpHttpRequest` implementation to the plugin side of things as default transport. This one can't be added to the libraries though, they know nothing about WP, so `CurlRemoteRequest` makes more sense there.